### PR TITLE
Clean battle interface TurnOrder class

### PR DESCRIPTION
**Pull request by Districh-ru**
This PR cleans the `TurnOrder` class.
In this PR the calculation of unit icons positions is done only when the units count is changed and on every redraw call.
The pointers to the units are still updated on every redraw call since they could be changed and currently there is no way to easy detect such change.

### CODE REVIEW

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -1091,42 +1080,42 @@ void Battle::TurnOrder::QueueEventProcessing( std::string & msg, const fheroes2:
...
-    number.draw( pos.x + 2, pos.y + 4, output );
+    number.drawInRoi( pos.x + 2, pos.y + 4, output, pos );
 
     if ( isCurrentUnit ) {
-        fheroes2::DrawRect( output, { pos.x, pos.y, turnOrderMonsterIconSize, turnOrderMonsterIconSize }, currentUnitColor );
+        fheroes2::DrawRect( output, pos, currentUnitColor );
     }
     else {
-        uint8_t color = 0;
+        uint8_t color = ARMY_COLOR_BLUE;
```
Comment (ihhub) [inline src/fheroes2/battle/battle_interface.cpp:1112]: Let's make the default color as **ARMY_COLOR_GRAY** which is a neutral army.

**Review by ihhub — COMMENTED** [review]
Hi @Districh-ru , I left one suggestion here. Could you please check it?

File: src/fheroes2/battle/battle_interface.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/battle/battle_interface.cpp]: Sure!

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for this cleanup!

**Pull request merged**